src/vl53_main.c

```diff
@@ -0,0 +1,270 @@
+            if (RightPreviousStatus == SOMEONE)     // need to check right zone as well ...
+                AllZonesCurrentStatus += 2;         // event in left zone has occured
+            LeftPreviousStatus = CurrentZoneStatus; // remember for next time
+        }
+    }
+    // Zone 1
+    else
+    {
+        if (CurrentZoneStatus != RightPreviousStatus)
+        {
+            AnEventHasOccured = 1; // event in left zone has occured
+            if (CurrentZoneStatus == SOMEONE)
+                AllZonesCurrentStatus += 2;
+            if (LeftPreviousStatus == SOMEONE)       // need to left right zone as well ...
+                AllZonesCurrentStatus += 1;          // event in left zone has occured
+            RightPreviousStatus = CurrentZoneStatus; // remember for next time
+        }
+    }
+
+    // if an event has occured
+    if (AnEventHasOccured)
+    {
+        if (PathTrackFillingSize < 4)
+            PathTrackFillingSize++;
+
+        // if nobody anywhere lets check if an exit or entry has happened
+        if ((LeftPreviousStatus == NOBODY) && (RightPreviousStatus == NOBODY))
+        {
+            // check exit or entry only if PathTrackFillingSize is 4 (for example 0 1 3 2) and last event is 0 (nobobdy anywhere)
+            if (PathTrackFillingSize == 4)
+            {
+                // check exit or entry. no need to check PathTrack[0] == 0 , it is always the case
+                if ((PathTrack[1] == 1) && (PathTrack[2] == 3) && (PathTrack[3] == 2))
+                    PeopleCount++; // This an entry
+                else if ((PathTrack[1] == 2) && (PathTrack[2] == 3) && (PathTrack[3] == 1))
+                    PeopleCount--; // This an exit
+            }
+            PathTrackFillingSize = 1;
+        }
+        else
+        {
+            // update PathTrack
+            // example of PathTrack update
+            // 0
+            // 0 1
+            // 0 1 3
+            // 0 1 3 1
+            // 0 1 3 3
+            // 0 1 3 2 ==> if next is 0 : check if exit
+            PathTrack[PathTrackFillingSize - 1] = AllZonesCurrentStatus;
+        }
+    }
+    return (PeopleCount);
+}
+
+// Function checks if the PeoleCount value is not faulty
+// E
+int Check_PeopleCount(uint16_t PeopleCount)
+{
+    if (PeopleCount & ((uint16_t)1 << 15)) // In Two's complement, MSB is set and the other bits are flipped when value is negative. Check if MSB is set.
+    {
+        set_Led(LED1, LED_ON); // Turn on green error led
+        if (PeopleCount != last_PeopleCount)
+        {   
+            send_Warning("Negative Value: Error in count!\r\n");
+            send_Warning("People Counter set to 0!\r\n");
+            PeopleCount = last_PeopleCount = 0;
+        }
+    }
+    return PeopleCount;
+}
+
+// These functions are only needed in debug mode
+#ifdef debug
+
+// Function displays amount of people that entered the room
+void display_peoplecounter(int PeopleCount)
+{
+    char buffer[50];

```

Comment:
Heb je hier zo'n grote buffer voor nodig?

Reply:
is alleen voor debug, wordt niet in project gebruikt verder